pkg/kubernetes/scripts/registry.js

```diff
@@ -163,6 +164,59 @@
                 return select().kind("Project");
             };
 
+            // TODO: invalidate cache on role changes
+            var sarCache = { };
+
+            $scope.subjectAccessReview = function subjectAccessReview(project, user, verb, resource) {
+                var result = false;
+
+                // if no project was given, check the first one as an approximation
+                // TODO: is that sufficient, or should we iterate over all
+                // projects? (much more expensive)
+                if (user && !project) {
+                    var projects = select().kind("Project");
+                    for (var p in projects) {
+                        project = projects[p].metadata.name;
+                        break;
+                    }
+                    if (!project)
+                        // we didn't init sufficiently far yet, defer to the
+                        // next call; or the user really has no projects
+                        return false;
```

Comment:
Do you have a hint how to do that in the HTML? I currently do
```html
+            <div id="docker-push-commands" ng-show="subjectAccessReview(null, settings.currentUser, 'update', 'imagestreamimages')">
```
so supposedly this needs to be wrapped into some deferring "only evaluate this once projects have been loaded". I'm currently lost how to approach this, do we do this somewhere else already? Thanks!